### PR TITLE
Add jsx linked editing

### PR DESCRIPTION
**Pull request by mjbvz**
For https://github.com/microsoft/TypeScript/issues/51832

### CODE REVIEW

**Review by lramos15 — DISMISSED** [review]

File: extensions/typescript-language-features/src/tsServer/server.ts [hunk 1]
```diff
@@ -495,6 +495,7 @@ export class SyntaxRoutingTsServer extends Disposable implements ITypeScriptServ
 		'format',
 		'formatonkey',
 		'docCommentTemplate',
+		'LinkedEditing'
```
Comment (DanielRosenwasser) [inline extensions/typescript-language-features/src/tsServer/server.ts:498]: I think this is outdated now

File: extensions/typescript-language-features/src/languageFeatures/jsxLinkedEditing.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+export function register(
+	selector: DocumentSelector,
+	client: ITypeScriptServiceClient
+) {
+	return conditionalRegistration([
+		requireMinVersion(client, JsxLinkedEditingSupport.minVersion),
+		requireSomeCapability(client, ClientCapability.Syntax),
+	], () => {
+		return vscode.languages.registerLinkedEditingRangeProvider(selector.semantic,
```
Comment (DanielRosenwasser) [inline extensions/typescript-language-features/src/languageFeatures/jsxLinkedEditing.ts:46]: Should this be `selector.syntax`?

File: extensions/typescript-language-features/src/languageFeatures/jsxLinkedEditing.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { DocumentSelector } from '../configuration/documentSelector';
+import { API } from '../tsServer/api';
+import * as typeConverters from '../typeConverters';
+import { ClientCapability, ITypeScriptServiceClient } from '../typescriptService';
+import { conditionalRegistration, requireMinVersion, requireSomeCapability } from './util/dependentRegistration';
+
+class JsxLinkedEditingSupport implements vscode.LinkedEditingRangeProvider {
```
Comment (DanielRosenwasser) [inline extensions/typescript-language-features/src/languageFeatures/jsxLinkedEditing.ts:13]: Nit - you currently don't limit this to any JSX-specific file, so maybe we rename this to `LinkedEditingSupport` and just leave a comment about the main use-case being JSX tags?

**Review by rebornix — APPROVED** [review]

**Review by DanielRosenwasser — COMMENTED** [review]
Probably fine as is, but I think I caught a minor issue using the semantic selector instead of the syntactic one.

File: extensions/typescript-language-features/src/languageFeatures/jsxLinkedEditing.ts (hunk 3 above)
Comment (mjbvz) [reply extensions/typescript-language-features/src/languageFeatures/jsxLinkedEditing.ts:13]: Thanks. Will follow up with these changes in a new PR

**Pull request merged**